GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -572,14 +572,22 @@ public Result CatFileGetType(string objectId)
             return this.InvokeGitAgainstDotGitFolder("cat-file -t " + objectId);
         }
 
-        public Result LsTree(string treeish, Action<string> parseStdOutLine, bool recursive, bool showAllTrees = false)
+        public Result LsTree(string treeish, Action<string> parseStdOutLine, bool recursive, bool showAllTrees = false, bool showDirectories = false)
         {
             return this.InvokeGitAgainstDotGitFolder(
-                "ls-tree " + (recursive ? "-r " : string.Empty) + (showAllTrees ? "-t " : string.Empty) + treeish,
+                "ls-tree " + (recursive ? "-r " : string.Empty) + (showAllTrees ? "-t " : string.Empty) + (showDirectories ? "-d " : string.Empty) + treeish,
                 null,
                 parseStdOutLine);
         }
 
+        public Result LsFiles(Action<string> parseStdOutLine, bool showSkipTreeBit = false)
+        {
+            return this.InvokeGitInWorkingDirectoryRoot(
+                "ls-files " + (showSkipTreeBit ? "-sv " : string.Empty),

```

Comment:
This is my fault but for `showSkipTreeBit` we can just use the `-v`.  The `-s` is to show the file mode, sha, and stage for the path.
```
C:\github\VFSForGit>git ls-files -sv Readme.md
H 100644 3f7e72a5b3c9eefdf993bd92a257e63f85daa878 0     Readme.md
```
vs
```
C:\github\VFSForGit>git ls-files -v Readme.md
H Readme.md
```